databases/week2/task4.sql

```diff
@@ -0,0 +1,59 @@
+CREATE DATABASE smoothie;
+
+USE smoothie;
+
+CREATE TABLE creator (
+	creator_id INT NOT NULL
+	,NAME VARCHAR(255)
+	,age_range ENUM
+	,Gender ENUM
+	,PRIMARY KEY (creator_id)
+	);
+
+CREATE TABLE smoothie (
+	smoothie_id INT NOT NULL
+	,NAME VARCHAR(255)
+	,category VARCHAR(255)
+	,ingredients INT NOT NULL
+	,creator INT NOT NULL
+	,FOREIGN KEY (category) REFERENCES colorType(category)
+	,FOREIGN KEY (ingredients) REFERENCES ingredient(ingredient_id)
+	,FOREIGN KEY (creator) REFERENCES creator(creator_id)
+	,PRIMARY KEY (smoothie_id)
+	);
+
+CREATE TABLE type (
+	type_id INT NOT NULL
+	,NAME VARCHAR(255)
+	,PRIMARY KEY (type_id)
+	,
+	);
+
+CREATE TABLE color (
+	color_id INT NOT NULL
+	,NAME VARCHAR(255)
+	,PRIMARY KEY (color_id)
+	,
+	);
+
+CREATE TABLE ingredient (
+	ingredient_id INT NOT NULL
+	,NAME VARCHAR(255)
+	,type_id INT NOT NULL
+	,color_id INT NOT NULL
+	,PRIMARY KEY (ingredient_id)
+	,FOREIGN KEY (type_id) REFERENCES type(type_id)
+	,FOREIGN KEY (color_id) REFERENCES color(color_id)
+	);
+
+CREATE TABLE colorType (
+	color_id INT NOT NULL
+	,type_id INT NOT NULL
+	,category VARCHAR(255)
+	,FOREIGN KEY (color_id) INT NOT NULL REFERENCES color(color_id)
+	,FOREIGN KEY (type_id) INT NOT NULL REFERENCES type(type_id)
```

Comment:
You've already defined your color_id  and type_id values in their tables (it means you don't need to repeat int not_null).

Reply:
Ok, got it :)